### PR TITLE
feat: create html extension

### PR DESCRIPTION
**Pull request by housed**
# Create HTML Extension

This module adds initial support for the [Alexa Web API for Games](https://developer.amazon.com/en-US/docs/alexa/web-api-for-games/understand-alexa-web-api-for-games.html)

## Description

This initial implementation of `@litexa/html` extension adds Litexa dev support for adding `mark` SSML tags to their speech outputs.

## Motivation and Context

To get the ball rolling for Litexa developers so that they can create even more compelling game experiences on Alexa (via the Alexa Web API).

## Testing

Unit and integration tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa-games/litexa/issues
[license]: https://www.apache.org/licenses/LICENSE-2.0
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: packages/litexa-html/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+This is useful for your HTML runtime in that, when it receives the Alexa response,
+it can use these `mark` tags for additional context before you finally render any
+visual & auditory content to the user.
+
+## Relevant Resources
+
+For more information, please refer to the official Alexa Web API for Games documentation:
+
+* [Alexa Web API 
+for Games](https://developer.amazon.com/en-US/docs/alexa/web-api-for-games/understand-alexa-web-api-for-games.html)
```
Comment (ayrc) [inline packages/litexa-html/README.md]: This link is going to the general ASK page and saying it's currently experiencing technical difficulties. I'm not sure if this is temporary though. The only link I have for this feature is the blog post: https://developer.amazon.com/blogs/alexa/post/1485655a-84d8-45ea-b396-9140bcd5d735/apply-for-the-alexa-web-api-for-games-developer-preview
It's possible you need to apply for the preview before getting access to the documentation?

File: packages/litexa-html/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```coffeescript
+launch
+  say "Hello, World!"
+  HTML.mark("screen:blue")
+  END
+```
+
+This will add a substring like `<mark name="screen:blue"/>` to your response's speech. 
+This is useful for your HTML runtime in that, when it receives the Alexa response,
+it can use these `mark` tags for additional context before you finally render any
```
Comment (ayrc) [inline packages/litexa-html/README.md]: Would reword such that it explains that the mark positions map to when they are fired off in the SSML output rendering.

File: packages/litexa/src/parser/handler.coffee [hunk 3]
```diff
@@ -542,6 +542,10 @@ handlerSteps.createFinalResult = (stateContext) ->
   if stateContext.directives? and stateContext.directives.length > 0
     response.directives = stateContext.directives
 
+  # last chance, see if the developer left a postprocessor to run here
+  if litexa.responsePostProcessor?
+    litexa.responsePostProcessor wrapper, stateContext
+
```
Comment (ayrc) [inline packages/litexa/src/parser/handler.coffee]: can you add a `@TODO` for documenting how to use this in the backdoor.md file?

File: packages/litexa-html/README.md (hunk 1 above)
Comment (housed) [reply packages/litexa-html/README.md]: Yeah, spoke to someone else and it does require special privileges for access. I'm thinking that we should add a note to the README that links to the initial blog post & explains that, no matter what, if you want to use Alexa.Presentation.HTML interface & see the docs, you have to sign up for the dev preview.

File: packages/litexa-html/README.md (hunk 2 above)
Comment (joseifert) [reply packages/litexa-html/README.md]: +1
HTML behavior can be controlled with HTML message directives, whereas marks can be used to trigger HTML events that interleave with SSML playback.
